OOP/OOP-3/TrainReservation.java

```diff
@@ -0,0 +1,322 @@
+		bookingInformation.add(Integer.toString(amount));
+		System.out.println("Successfull Booking");
+		break;
+		case 3:System.out.println("Enter Bank Name");
+		String bankName=sc.next();
+		bookingInformation.add(bankName);
+		System.out.println("Enter user name:");
+		String userName=sc.next();
+		bookingInformation.add(userName);
+		System.out.println("Enter password:");
+		String password=sc.next();
+		bookingInformation.add(password);
+		System.out.println("Successfull Booking");
+		}
+		return bookingInformation;
+	}
+}
+//this class is used to hold personal information of user 
+class PersonalInformation
+{
+	private String name;
+	public PersonalInformation(String name)
+	{
+		this.name=name;
+	}
+	public void setName(String name)
+	{
+		this.name=name;
+	}
+	public String getName()
+	{
+		return name;
+	}
+}
+class TrainBooking
+{
+	Scanner sc;
+	int flag=0;
+	ArrayList<String> paymentInformation;
+	String trainNumber,from,to,trainType,name;
+	int numberOfSeats;
+	ArrayList<String> trainInfo;
+	TrainBooking(String trainNumber,int numberOfSeats,ArrayList<String> trainInfo,String from,String to,String trainType,String name)
+	{
+		this.trainNumber=trainNumber;
+		this.numberOfSeats=numberOfSeats;
+		this.trainInfo=trainInfo;
+		this.from=from;
+		this.to=to;
+		this.trainType=trainType;
+		this.name=name;
+	}
+	//this method is used for printing the booking information
+	public void booking()
+	{
+		try
+		{
+			for(int i=0;i<trainInfo.size();i++)
+			{
+				String temp[]=trainInfo.get(i).split(",");
+				if(temp[0].equalsIgnoreCase(trainNumber) && Integer.parseInt(temp[3])>=numberOfSeats)
+				{
+					PersonalInformation pInformation=new PersonalInformation(name);
+					System.out.print("Payment Detail");
+					Payment payment=new Payment(temp);
+					paymentInformation=payment.paymentType();
+					System.out.println("Deatail:"+name+paymentInformation);
+					flag++;
+					TrainChart trainChart=new TrainChart(trainType, trainInfo);
+					trainChart.updatedChart(numberOfSeats,trainNumber);
+				}
+			}
+		}catch(Exception e){}
+		if(flag==0)
+		{
+			System.out.println("sorry no seats available");
+		}
+		//return paymentInformation;
+	}
+}
+//main class for train reservation
+public class TrainReservation  
+{
+	public static void main(String args[])throws FileNotFoundException,NullPointerException,IOException
+	{
+		Scanner sc=new Scanner(System.in);
+		BufferedReader bfTrain;
+		String line;
+		String name,trainType,from,to;
+		
+		while(true)
+		{
+			System.out.println("1:login\n2:exit");
+			int hold=sc.nextInt();
+			switch(hold)
+			{
+			case 1:	ArrayList<String> trainInfo=new ArrayList<String>();
+				    FileReader finGoodsTrain=new FileReader("Goods.txt");
+				    FileReader finPasangerTrain=new FileReader("Passanger.txt");
+				    System.out.println("Enter Name");
+					name=sc.next();
+					TrainChart trainchart;
+					System.out.println("Enter Train Type Passanger/Goods");
+					trainType=sc.next();
```

Comment:
ACTION:Use try catch here because if user enters in incorrect input we have to give proper message to him.